README.md

```diff
@@ -44,10 +45,162 @@ thrift_files: Mix.Utils.extract_files(["thrift"], [:thrift])
 By default, the generated source files will be written to the `lib` directory,
 but you can change that using the `thrift_output` option.
 
+## Working with Thrift
+
+The following examples use the following thrift definition:
+
+```thrift
+namespace elixir Thrift.Test 
+exception UserNotFound {
+  1: string message
+}
+struct User {
+  1: i64 id,
+  2: string username,
+  3: string first_name,
+  4: string last_name
+}
+
+service UserService {
+  1: bool ping(),
+  2: User get_user_by_id(1: i64 user_id) throws (1: UserNotFound unf),
+  3: boolean deleteUser(1: i64 userId),
+}
+```
+
+#### The generated code will be placed in the following modules:
```

Comment:
We haven't documented the compiler options in the README... that could be a followup PR?